### PR TITLE
Fix two issues in SelectAndPollTests

### LINKED ISSUE #4686: SelectRead_Multiple_Success failing in CI
http://dotnet-ci.cloudapp.net/job/dotnet_corefx_mac_debug_tst/140/console
```
04:51:52    System.Net.Sockets.Tests.SelectAndPollTests.SelectRead_Multiple_Success [FAIL]
04:51:52       Assert.Equal() Failure
04:51:52       Expected: 2
04:51:52       Actual:   1
04:51:52       Stack Trace:
04:51:52             at System.Net.Sockets.Tests.SelectAndPollTests.SelectRead_Multiple_Success()
```

### PR DESCRIPTION
**Pull request by stephentoub**
1. The timeout value used in the tests was written assuming the unit was milliseconds, but microseconds is actually used.  This meant we were passing in a timeout of 5 milliseconds instead of 5 seconds.  Changed to use the right value.

2. The SelectRead_Multiple_Success test attempts to use Select as a form of "when all", passing in two sockets and expecting that it'll find both have data available.  But the data is available asynchronously from the send, and Select returns when at least one of the sockets is readable.  As a result, there's a race condition that results in the test failing sporadically: on my machine, it fails ~15 out of every 1000 runs.  I've fixed the test by looping until either the timeout is reached or we find both sockets available, whichever comes first.

Fixes #4686
cc: @pgavlin, @CIPop, @davidsh

### CODE REVIEW

File: src/System.Net.Sockets.Legacy/tests/FunctionalTests/SelectAndPollTests.cs [hunk 1]
```diff
@@ -15,19 +16,19 @@ namespace System.Net.Sockets.Tests
     public class SelectAndPollTests
     {
         const int SelectTimeout = 100;
-        const int SelectSuccessTimeout = 5000;
+        const int SelectSuccessTimeoutMicroseconds = 5000000;
```
Comment (davidsh) [inline src/System.Net.Sockets.Legacy/tests/FunctionalTests/SelectAndPollTests.cs]: nit: add comment at end of line something like
```c#
// 5 seconds
```

All those zeroes are hard to visually parse.

Comment (stephentoub) [inline src/System.Net.Sockets.Legacy/tests/FunctionalTests/SelectAndPollTests.cs]: Fixed.

File: src/System.Net.Sockets.Legacy/tests/FunctionalTests/SelectAndPollTests.cs [hunk 2]
```diff
@@ -15,19 +16,19 @@ namespace System.Net.Sockets.Tests
     public class SelectAndPollTests
     {
         const int SelectTimeout = 100;
-        const int SelectSuccessTimeout = 5000;
+        const int SelectSuccessTimeoutMicroseconds = 5*000*000; // 5 seconds
```
Comment (justinvp) [inline src/System.Net.Sockets.Legacy/tests/FunctionalTests/SelectAndPollTests.cs]: `5*1000*1000`?

Comment (stephentoub) [inline src/System.Net.Sockets.Legacy/tests/FunctionalTests/SelectAndPollTests.cs]: Ugh! :smile:

**Comment by pgavlin** [comment]
D'oh. Good catch on the race. LGTM.

**Comment by stephentoub** [comment]
Test Innerloop CentOS7.1 Release Build and Tes please ("The process cannot access the file")

**Comment by stephentoub** [comment]
Test Innerloop CentOS7.1 Release Build and Test please ("The process cannot access the file")

**Pull request merged**